### PR TITLE
Feature/remove user from group

### PR DESCRIPTION
**Pull request by mwnicks**
### What

Add a remove user from group endpoint.

### How to review

Ensure all the unit and feature test are passing
Send a POST request to /v1/groups/role-publisher/members/ca83ca4c-0495-4254-a131-57ada4b1fc70, with body:

You should receive a 200 response with body that looks like:
{
"name": "role-publisher",
"description": "The publishers",
"precedence": 1,
"created": "2021-07-08T16:59:24.---Z"
"members": []
}

### Who can review

Anyone but me

### CODE REVIEW

File: api/groups.go [hunk 1]
```diff
@@ -66,3 +66,51 @@ func (api *API) AddUserToGroupHandler(ctx context.Context, w http.ResponseWriter
 
 	return models.NewSuccessResponse(jsonResponse, http.StatusOK, nil), nil
 }
+
+//RemoveUserFromGroupHandler adds a user to the specified group
+func (api *API) RemoveUserFromGroupHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	vars := mux.Vars(req)
+	group := models.Group{Name: vars["id"]}
```
Comment (mike-kelly-methods) [inline api/groups.go:73]: Are we always certain that the`.../{id}/...`  param passed to the handler represents a valid group?

File: api/groups_test.go [hunk 2]
```diff
@@ -257,3 +258,243 @@ func TestAddUserToGroupHandler(t *testing.T) {
...
+						Group: getGroupData,
+					}
+					return group, nil
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					return &cognitoidentityprovider.ListUsersInGroupOutput{}, nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					So(successResponse, ShouldNotBeNil)
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
```
Comment (mike-kelly-methods) [inline api/groups_test.go:301]: This should be `http.StatusAccepted` for a `202` (or maybe the comment at the top of this test item is incorrect?)

File: api/groups_test.go [hunk 3]
```diff
@@ -257,3 +258,243 @@ func TestAddUserToGroupHandler(t *testing.T) {
...
+					So(errorResponse, ShouldBeNil)
+				},
+			},
+			// Cognito 404 response - user not found
```
Comment (mike-kelly-methods) [inline api/groups_test.go:305]: Looking at the response, and the swagger doc, I think this should be `400`

Comment (mwnicks) [reply api/groups_test.go:305]: This comment is describing the response from Cognito not the response from the API.
Would probably be less confusing to leave the Cognito status code out of the comment

File: api/groups_test.go (hunk 2 above)
Comment (mwnicks) [reply api/groups_test.go:301]: It was a 202 response, I changed it and must have missed updating the comment.
Will update now.

File: api/groups_test.go (hunk 3 above)
Comment (mwnicks) [reply api/groups_test.go:305]: I think I've misunderstood your comment here.

File: api/groups_test.go (hunk 2 above)
Comment (mike-kelly-methods) [reply api/groups_test.go:301]: 👍

File: api/groups_test.go (hunk 3 above)
Comment (mike-kelly-methods) [reply api/groups_test.go:305]: My bad - its the 404 from Cognito

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Pull request merged**